### ISSUE TITLE
Perl errors received when trying to use on a git repo

### ISSUE THREAD

**Issue by GodGinrai** [AlDanial/cloc#328]
I'm getting some cryptic perl errors when trying to run your tool on a commit in a rep as described.

```
[14:47 onyx/]: cloc 919ea0da6f05296aa5ca0b5f45f0b439ac4a906a        (master|✔)
Use of uninitialized value $A2 in concatenation (.) or string at C:\Users\user_name\AppData\Roaming\npm/node_modules/cloc/lib/cloc line 4486.
Can't use string ("919ea0da6f05296aa5ca0b5f45f0b439"...) as an ARRAY ref while "strict refs" in use at C:\Users\user_name\AppData\Roaming\npm/node_modules/cloc/lib/cloc line 4489.
```

Repository used is enyojs/onyx@919ea0da6f05296aa5ca0b5f45f0b439ac4a906a

**Comment by AlDanial** [comment]
Which version of cloc is this?  If not the latest, 1.78, please try that one first.  If it is 1.78, what is the output when running with the verbose switch ``-v``?

**Comment by GodGinrai** [comment]
v1.78, installed via npm in a Windows environment.  The `-v` flag does not give any additional output. (as you can see from @u55's comment)

**Comment by AlDanial** [comment]
This certainly reveals a gaping hole in my test suite :/  Here's the before & after code on the fix:
Before
<pre>
 4485     } else {
 4486         push @File_Set, "$A1 $A2";
 4487     }
</pre>

After
<pre>
 4485     } else {
 4486         if (defined $A2) {
 4487             push @{$File_Set[0]}, "$A1 $A2";
 4488         } else {
 4489             push @{$File_Set[0]}, "$A1";
 4490         }
 4491     }
</pre>

In other words, replace the one line at 4486 with the new if-statement, lines 4486-4490 in the After block, above.  I'll push thix fix later today, but you should be able to edit your cloc file directly if you want.

**Comment by AlDanial** [comment]
I'll definitely add more tests because of this.  Quite embarrassing.
